// See README.txt for information and build instructions.
//
// Note: START and END tags are used in comments to define sections used in
// tutorials.  They are not part of the syntax for Protocol Buffers.
//
// To get an in-depth walkthrough of this file and the related examples, see:
// https://developers.google.com/protocol-buffers/docs/tutorials

// [START declaration]
syntax = "proto3";
package tutorial;

import "google/protobuf/timestamp.proto";
// [END declaration]

// [START java_declaration]
option java_package = "com.example.tutorial";
option java_outer_classname = "AddressBookProtos";
// [END java_declaration]

// [START csharp_declaration]
option csharp_namespace = "Google.Protobuf.Examples.AddressBook";
// [END csharp_declaration]

// [START messages]
message Person {
  string name = 1;
  int32 id = 2;  // Unique ID number for this person.
  string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    string number = 1;
    PhoneType type = 2;
  }

  repeated PhoneNumber phones = 4;

  google.protobuf.Timestamp last_updated = 5;

  int64 new_int64 = 6;
  double new_double = 7;
  float new_float = 8;
  uint32 new_uint32 = 9;
  uint64 new_uint64 = 10;
  sint32 new_sint32 = 11;
  sint64 new_sint64 = 12;
  fixed32 new_fixed32 = 13;
  fixed64 new_fixed64 = 14;
  sfixed32 new_sfixed32 = 15;
  sfixed64 new_sfixed64 = 16;
  bool new_bool = 17;

}

// Our address book file is just one of these.
message AddressBook {
  repeated Person people = 1;
}
// [END messages]

